sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/ThriftServerWithSparkContextSuite.scala

```diff
@@ -149,6 +149,64 @@ trait ThriftServerWithSparkContextSuite extends SharedThriftServer {
       }
     }
   }
+
+  test("formats of error messages") {
+    val sql = "select 1 / 0"
+    withCLIServiceClient() { client =>
+      val sessionHandle = client.openSession(user, "")
+      val confOverlay = new java.util.HashMap[java.lang.String, java.lang.String]
+      val exec: String => OperationHandle = client.executeStatement(sessionHandle, _, confOverlay)
+
+      exec(s"set ${SQLConf.ANSI_ENABLED.key}=true")
+      exec(s"set ${SQLConf.ERROR_MESSAGE_FORMAT.key}=${ErrorMessageFormat.PRETTY}")
+      val e1 = intercept[HiveSQLException](exec(sql))
+      // scalastyle:off line.size.limit
+      assert(e1.getMessage ===
+        """Error running query: [DIVIDE_BY_ZERO] Division by zero. Use `try_divide` to tolerate divisor being 0 and return NULL instead. If necessary set "spark.sql.ansi.enabled" to "false" to bypass this error.
+          |== SQL(line 1, position 8) ==
+          |select 1 / 0
+          |       ^^^^^
+          |""".stripMargin)
+
+      exec(s"set ${SQLConf.ERROR_MESSAGE_FORMAT.key}=${ErrorMessageFormat.MINIMAL}")
+      val e2 = intercept[HiveSQLException](exec(sql))
+      assert(e2.getMessage ===
+        """Error running query: {
+          |  "errorClass" : "DIVIDE_BY_ZERO",
+          |  "sqlState" : "22012",
+          |  "messageParameters" : {
+          |    "config" : "\"spark.sql.ansi.enabled\""
+          |  },
+          |  "queryContext" : [ {
+          |    "objectType" : "",
+          |    "objectName" : "",
+          |    "startIndex" : 8,
+          |    "stopIndex" : 12,
+          |    "fragment" : "1 / "
```

Comment:
is it a bug? it should be `1 / 0`

Reply:
Probably

Reply:
is this bug thrift-server only? if yes we can probably create a ticket and ask someone who knows thriftserver better to fix it.

Reply:
It is not thrift-server specific:
```scala
    withSQLConf(SQLConf.ANSI_ENABLED.key -> "true") {
      val e = intercept[SparkArithmeticException] {
        sql("select 1 / 0").collect()
      }
      println("'" + e.getQueryContext()(0).fragment() + "'")
    }

'1 / '
```

Reply:
Seems like the issue in:
```scala
  override lazy val fragment: String = {
    if (!isValid) {
      ""
    } else {
      sqlText.get.substring(originStartIndex.get, originStopIndex.get)
    }
  }
```
`originStopIndex.get` points to the last char of the fragment but `substring()` takes a sub-string up to `endIndex - 1`.

Let me open a PR and fix this separately.

Reply:
Here is the JIRA for the issue: https://issues.apache.org/jira/browse/SPARK-40136

Reply:
I have opened the PR w/ a bug fix: https://github.com/apache/spark/pull/37566